pyomo/core/plugins/transform/bounds_to_vars.py

```diff
@@ -0,0 +1,77 @@
+"""Transformation to convert explicit bounds to variable bounds."""
+from __future__ import division
+
+from pyomo.core.base.constraint import Constraint
+from pyomo.core.base.var import Var
+from pyomo.core.kernel.numvalue import value
+from pyomo.core.plugins.transform.hierarchy import IsomorphicTransformation
+from pyomo.util.plugin import alias
+import pyomo.core.base.expr as EXPR
+from pyomo.repn.canonical_repn import generate_canonical_repn
+
+
+class ConstraintToVarBoundTransform(IsomorphicTransformation):
+    """Change constraints to be a bound on the variable.
+
+    Looks for constraints of form k*v + c1 <= c2. Changes bound on v to match
+    (c2 - c1)/k if it results in a tighter bound. Also does the same thing for
+    lower bounds.
+
+    """
+
+    alias('core.constraints_to_var_bounds', doc=__doc__)
+
+    def __init__(self, *args, **kwargs):
+        """Initialize the transformation."""
+        super(ConstraintToVarBoundTransform, self).__init__(*args, **kwargs)
+
+    def _create_using(self, model):
+        """Create new model, applying transformation."""
+        m = model.clone()
+        self._apply_to(m)
+        return m
+
+    def _apply_to(self, model):
+        """Apply the transformation to the given model."""
+        m = model
+
+        for constr in m.component_data_objects(ctype=Constraint,
+                                               active=True,
+                                               descend_into=True):
+            # Check if the constraint is k * x + c1 <= c2 or c2 <= k * x + c1
+            if constr.body.polynomial_degree() == 1:
+                repn = generate_canonical_repn(constr.body)
+                if repn.variables is not None and len(repn.variables) == 1:
+                    var = repn.variables[0]
+                    const = repn.constant if repn.constant is not None else 0
+                    coef = repn.linear[0]
```

Comment:
DOH.  I missed that.  Yes - the `__future__` will cover it.  [So you are seeing a style difference ... instead of remembering to start every script with the division import, I just trained myself to explicitly convert something to float]